File: Iteration2-Python/src/Person.py

from abc import ABC,abstractmethod

import logging
logger = logging.getLogger(__name__)    # to ensure that the logger is named after the module
class Person(ABC):

   def __init__(self, id, name, e_mail, Password, controller_instance):
         self.__id : int = id
         self.__name : str = name
         self.__e_mail : str = e_mail
         self.__password : str = Password
         self.__controller_instance = controller_instance


   #def check_password(self, password):
   #   return password.__eq__(self.__password)

   def get_password(self) -> str:
      return self.__password
   
   def get_e_mail(self) -> str:
      return self.__e_mail
   
   def get_id(self) -> int:
      return self.__id
   
   def get_name(self) -> str:
      return self.__name
            
   def get_controller_instance(self):
      return  self.__controller_instance
      
   @abstractmethod
   def get_json(self):
      pass
   
   @abstractmethod
   def main_menu(self) -> None:
      pass

File: Iteration1-Java/src/main/java/cse3063group3/Person.java
package cse3063group3;


public abstract class Person {
    private int id;
    private String name;
    private String eMail;
    private String password;
    private CourseRegistrationSystem controllerInstance;

    
    /**
     * @param id ID of person.
     * @param name Name of person.
     * @param eMail E-Mail address of person.
     * @param password Password of person.
     * @param controllerInstance Referance to the controller instance.
     */
    public Person(int id, String name, String eMail, String password) {
        this.id = id;
        this.name = name;
        this.eMail = eMail;
        this.password = password;
        this.controllerInstance = CourseRegistrationSystem.getCourseRegistrationSystemInstance();
    }

    public boolean checkPassword(String password){
        return password.equals(this.password);
    }
    public String getPassword(){
        return this.password;
    }
    public String getEMail(){
        return this.eMail;
    }

    public int getId(){
        return (this.id);
    }

    public String getName(){
        return (this.name);
    }

    /**
     * <pre>
     * Constructs the logic and main text body for a Person.
     * Should generally be called right after successful login.
     * </pre>
     */
    public abstract void mainMenu();

    /**
     * @return Referance to the controller instance.
     */
    public CourseRegistrationSystem getControllerInstance(){
        return this.controllerInstance;
    }
    
}
    